### DISCUSSION TITLE
Pointer Events

### DISCUSSION THREAD

**Discussion by dennemark** [brianzinn/react-babylonjs#279] [Ideas]
Currently we have pointerDragBehaviour or useClick useHover events as well as imperative code like scene.onPointerX events.
Looking at r3f, it seems to pass canvas events onto their meshes allowing to write code that is similar to react-dom events: <mesh onPointerDown... />
I am wondering if we could/should wrap our scene or canvas events in some similar api? I could imagine it to be quite useful.
It might just have some performance implications, since meshes with such events would need to be pickable.
If there is interest, I hope I can find some time to look into it.
https://github.com/pmndrs/react-three-fiber/blob/master/packages/fiber/src/core/events.ts

Edit:
Looking at useClick, using ActionManager actually seems to be the right thing to use.
https://github.com/brianzinn/react-babylonjs/blob/master/packages/react-babylonjs/src/hooks/utilityHooks.ts

**Comment by brianzinn** [comment]
that's a great idea.  would be really good and useful feature to have that easier to connect declaratively without resorting to hooks and `useRef`, etc.
